### PR TITLE
Dynamic cluster - remove node tests

### PR DESCRIPTION
**Pull request by tohar-ayash**
Signed-off-by: tohar-ayash <toharayash1@gmail.com>

### CODE REVIEW

File: test/cluster/dynamic_cluster_remove_test.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+		members = append(members[:1], members[2:]...)
+		nodes = append(nodes[:1], nodes[2:]...)
+	case 2:
+		members = members[:len(members)-1]
+		nodes = nodes[:len(nodes)-1]
+	}
+	return members, nodes
+}
+
+func removeServer(t *testing.T, leaderServer *setup.Server, c *setup.Cluster, removeIndex int, activeServers []int) int {
```
Comment (tock-ibm) [inline test/cluster/dynamic_cluster_remove_test.go:29]: The order of the nodes and members is not necessarily the same as the index in the servers array.
Use the node ID instead.
In addition, it seems to me it makes sense to unify or reuse the `removeServerFromMembersAndNodes` above, no?

File: test/cluster/dynamic_cluster_remove_test.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	follower2 := (leaderIndex + 2) % 3
+	newLeaderIndex := -1
+	require.Eventually(t, func() bool {
+		newLeaderIndex = c.AgreedLeader(t, follower1, follower2)
+		return newLeaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	statusResponseEnvelope, err = c.Servers[newLeaderIndex].QueryClusterStatus(t)
+	require.Equal(t, 2, len(statusResponseEnvelope.GetResponse().Nodes))
+	require.Equal(t, 2, len(statusResponseEnvelope.GetResponse().Active))
```
[the remark this answers is not in the document]
Comment (tohar-ayash) [reply test/cluster/dynamic_cluster_remove_test.go:123]: done

File: test/cluster/dynamic_cluster_remove_test.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+		nodes = nodes[1:]
+	case 1:
+		members = append(members[:1], members[2:]...)
+		nodes = append(nodes[:1], nodes[2:]...)
+	case 2:
+		members = members[:len(members)-1]
+		nodes = nodes[:len(nodes)-1]
+	}
+	return members, nodes
+}
```
[the remark this answers is not in the document]
Comment (tohar-ayash) [reply test/cluster/dynamic_cluster_remove_test.go:27]: done

**Review by tock-ibm — APPROVED** [review]
@tohar-ayash I am going to approve this, but:
1. We need to open an issue for "GetClusterStatus" on a removed node.
2. One more test case: 
    - start a 5 node cluster.
    - shutdown node 5
    - remove node 5 from cluster 
    - restart node5, make sure it does not rejoin the cluster

**Pull request merged**